src/Main.purs

```diff
@@ -159,47 +172,20 @@ handleAction action = do
           (HTMLDocument.toEventTarget document)
           (map (HandleScroll id) <<< WE.fromEvent)
       handleAction Clear
-    Clear -> do
-      let
-        plots = map (\plot -> plot { expression = Nothing }) state.plots
-      drawCommands <- lift $ computePlots state.input.size state.bounds plots
-      H.put state { input { operations = drawCommands }, plots = plots }
-    HandleExpressionInput (Parsed id expression text) -> do
-      let
-        plots = map (updatePlot id expression text) state.plots
-      drawCommands <- lift $ computePlots state.input.size state.bounds plots
-      H.put state { input { operations = drawCommands }, plots = plots }
-    Pan direction -> do
-      let
-        newBounds = panBounds state.bounds direction
-      drawCommands <- lift $ computePlots state.input.size newBounds state.plots
-      H.put state { input { operations = drawCommands }, bounds = newBounds }
-    Zoom isZoomIn -> do
-      let
-        newBounds = zoomBounds state.bounds isZoomIn
-      drawCommands <- lift $ computePlots state.input.size newBounds state.plots
-      H.put state { input { operations = drawCommands }, bounds = newBounds }
-    ResetBounds -> do
-      drawCommands <- lift $ computePlots state.input.size initialBounds state.plots
-      H.put state { input { operations = drawCommands }, bounds = initialBounds }
     HandleScroll _ event -> do
       let
         changeInY = WE.deltaY event
       when (changeInY /= 0.0) do
         H.liftEffect $ E.preventDefault (WE.toEvent event)
         handleAction $ Zoom (changeInY < 0.0)
-    HandleCanvas (Dragged delta) -> do
-      let
-        newBounds = panBoundsByVector state.input.size state.bounds delta
-      drawCommands <- lift $ computePlots state.input.size newBounds state.plots
-      H.put state { input { operations = drawCommands }, bounds = newBounds }
-    AddPlot -> do
-      let
-        plots = state.plots <> [ newPlot (1 + length state.plots) ]
-      H.put state { plots = plots }
-    HandleBoundsInput (Updated newBounds) -> do
-      drawCommands <- lift $ computePlots state.input.size newBounds state.plots
-      H.put state { input { operations = drawCommands }, bounds = newBounds }
+    StartRobust id robustCommands bounds -> do
+      _ <- lift $ lift $ delay $ Milliseconds 2000.0 -- TODO Remove artifical delay
+      robustDrawCommands <- lift $ computePlots state.input.size robustCommands
+      handleAction $ EndRobust id robustDrawCommands
```

Comment:
This may work, but it is not what I had in mind when I suggested to raise an action.  I wonder whether it would behave even better if EndRobust would be raised in some Halogen native way, so that it is put into the queue of actions and handleAction would be called for this even by Halogen.  Something to investigate and test but I think it is not necessary to implement as part of this PR.

Reply:
When looking at all the halogen examples this was the only way that I saw an action trigger another action. I will have a look into it though.

Reply:
OK, thank you.  If there is a better way, it could save us some headaches when we do the more complicated asynchronous stuff.